deployability/modules/testing/testing.py

```diff
@@ -22,21 +23,49 @@ def run(cls, payload: InputPayload) -> None:
             payload (InputPayload): The payload containing the test parameters.
         """
         payload = InputPayload(**dict(payload))
-        inventory = Inventory(**Utils.load_from_yaml(payload.inventory))
-        logger.info(f"Running tests for {inventory.ansible_host}")
         extra_vars = cls._get_extra_vars(payload).model_dump()
+        print(payload)
+
+        targets = {}
+        dependencies = {}
+        extra_vars['hosts_ip'] = []
+
+        # Process targets and dependencies
+        for path_type, paths_list in [("targets", payload.targets), ("dependencies", payload.dependencies)]:
+            for path in paths_list:
+                dictionary = eval(path)
+                inventory = Inventory(**Utils.load_from_yaml(', '.join(dictionary.values())))
+                extra_vars['hosts_ip'].extend([inventory.ansible_host] if path_type == "targets" else [])
+                logger.info(f"Running tests for {(inventory.ansible_host)}") if path_type == "targets" else logger.info(f"Dependencies {inventory.ansible_host}")
+                if path_type == "targets":
+                    targets.update(dictionary)
+                else:
+                    dependencies.update(dictionary)
+
+        extra_vars['targets'] = json.dumps(targets).replace('"', "")
+        extra_vars['dependencies'] = json.dumps(dependencies).replace('"', "")
+
+        # Set extra vars
+        extra_vars['local_host_path'] = str(Path(__file__).parent.parent.parent)
+        extra_vars['current_user'] = os.getlogin()
+
         logger.debug(f"Using extra vars: {extra_vars}")
-        dependencies_dict = {}
-        for dependency in extra_vars['dependencies']:
-            dependency = ast.literal_eval(dependency)
-            dependencies_dict.update(dependency)
-        extra_vars['dependencies'] = dependencies_dict
-        ansible = Ansible(ansible_data=inventory.model_dump())
-        cls._setup(ansible, extra_vars['working_dir'])
+
+        ansible = None
```

Comment:
The `ansible` var does not require to be defined previous to its 'population' as it will always pass through the line `58`.

Reply:
Done https://github.com/wazuh/wazuh-qa/pull/4998/commits/a9db1da1940514178f94256e2d095d693bfc3220

Reply:
Done in https://github.com/wazuh/wazuh-qa/commit/d05be9a41f9aa224b48f5fb8208a042f2e9aeb7f